### PR TITLE
feat(Modal): add help section to the modal header

### LINKED ISSUE #4883: Add optional Help button to modal header
Includes an optional help button to the left of the Close button. Core issue here: https://github.com/patternfly/patternfly/issues/3474

For keyboard interaction, the Close button should continue to have focus when the user opens the modal. @jessiehuff will further investigate overall keyboard navigation when the help button is present.

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #4883

### CODE REVIEW

**Review by dlabrecq — DISMISSED** [review]
I see the help icon in the example, but it's not functional. I assume this would launch a popover or tooltip? It would be helpful to see how that pattern is expected to behave in the modal.

**Review by mcarrano — COMMENTED** [review]
@mcoker @mceledonia I thought we agreed that the Help button would be right-aligned and grouped with the Close button. Am I mistaken about that?

**Comment by kmcfaul** [comment]
@dlabrecq You are right that would be very helpful. I made the help a popover for the example!

**Comment by mcoker** [comment]
@mcarrano I have a core issue here to right align the help icon https://github.com/patternfly/patternfly/pull/3603

It's CSS only so once it's pulled in to react, the alignment should be fixed.

**Review by redallen — APPROVED** [review]
Code LGTM!

File: packages/react-core/src/components/Modal/examples/Modal.md [hunk 1]
```diff
@@ -793,3 +794,73 @@ class WithDropdown extends React.Component {
...
+        <Modal
+          title="Simple modal header"
+          help={
+            <Popover
+              headerContent={<div>Help Popover</div>}
+              bodyContent={
+                <div>
+                  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam id feugiat augue, nec fringilla
+                  turpis.
+                  <Button onClick={() => console.log('click')}>click</Button>
```
Comment (tlabaj) [inline packages/react-core/src/components/Modal/examples/Modal.md:835]: I don't know that we need this button in the example.  It looks a little off to me.  @mcarrano can you take a look at the popover content.  I am not sure that the `click` button would follow design recommendations.

**Review by tlabaj — COMMENTED** [review]
Other than waiting for the core bump to replace that inline style and the question I had about the `click` button in the popover, it looks good.

File: packages/react-core/src/components/Modal/examples/Modal.md (hunk 1 above)
Comment (kmcfaul) [reply packages/react-core/src/components/Modal/examples/Modal.md:835]: I will remove it.

Also, there is no actual `pf-c-modal-box__header-help` class, it doesn't do anything. It is a placeholder class to indicate the div.

**Review by mcarrano — APPROVED** [review]
Looks great. Thanks @kmcfaul !

**Review by redallen — APPROVED** [review]

**Review by jessiehuff — APPROVED** [review]
Small nit, but overall really great! 😄 When navigating by keyboard, when you try to hit escape to close the popover, you close the entire modal. We likely need to stop event propagation in the example (when the popover is open, escape should close the popover, when the popover is closed, escape should close the modal). Would be great to add this small aspect, but I'm ok with us adding that later!

**Review by mcoker — APPROVED** [review]
I ❤️ it!

**Pull request merged**